rasa/core/agent.py

```diff
@@ -411,7 +411,7 @@ def is_ready(self, allow_nlu_only: bool = False):
         )
 
     async def parse_message_using_nlu_interpreter(
-        self, message_data: Text
+        self, message_data: Text, tracker: DialogueStateTracker = None
```

Comment:
how come you only apply this to `parse_message_using_nlu_interpreter`? what about the dialogue methods, i.e. `handle_message()` and `handle_text()`?

Reply:
ok, went back and added types to functions I've touched. As I see it, Rasa codebase isn't 100% refactored to use argument types, and further refactoring is possible, but outside of the scope of this PR.